src/esphomelib/sensor/tx20_component.cpp

```diff
@@ -0,0 +1,169 @@
+#include "esphomelib/defines.h"
+
+#ifdef USE_TX20
+
+#include "esphomelib/log.h"
+#include "esphomelib/sensor/tx20_component.h"
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace sensor {  // namespace
+
+static const char *TAG = "sensor.tx20";
+static const uint8_t MAX_BUFFER_SIZE = 41;
+static const uint16_t TX20_MAX_TIME = MAX_BUFFER_SIZE * 1220 + 5000;
+static const uint16_t TX20_BIT_TIME = 1200;
+static const char* DIRECTIONS[] = {"N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE",
+                                    "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"};
+
+TX20Component::TX20Component(const std::string &wind_speed_name, const std::string &wind_direction_degrees_name,
+                             GPIOPin *pin)
+    : Component(),
+      wind_speed_sensor_(new TX20WindSpeedSensor(wind_speed_name)),
+      wind_direction_degrees_sensor_(new TX20WindDirectionDegreesSensor(wind_direction_degrees_name)),
+      pin_(pin) {
+
+}
+
+void TX20Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TX20...");
+  TX20Component::buffer_ = new uint16_t[MAX_BUFFER_SIZE];
```

Comment:
buffer_ should be allocated in setup (or constructor). If it is allocated globally every user will lose quite a bit of RAM even if they don't use this integration (plus the allocator on the ESPs is sometimes a bit flakey when it is activated pre user code)

Reply:
If this is the case it makes sense. I even had the code like that, but was under the impression that the code inside of the USE_TX20 only takes effect when it is in use and therefore moved it down to save a line.